app/test_main.py

```diff
@@ -1 +1,43 @@
-# write your code here
+from unittest import mock
+import pytest
+
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_call(mocked: mock.MagicMock) -> None:
+    can_access_google_page("https://www.google.com/webhp?client=firefox-b-d")
+    mocked.assert_called_once()
```

Comment:
you should add a descriptive message here in case if test fails - same in function below